### PR TITLE
AWS S3 config details added

### PR DESCRIPTION
**Pull request by anantshri**
# Description

It was interesting to note that since presigned urls are used buckets dont need to be exposed publically. this was an interesting change compared to other mastodon specific s3 bucket guides hence documented here for correct directions.

## Checklist

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.

### CODE REVIEW

**Comment by tsmethurst** [comment]
This looks useful, but it's also rather AWS-centric, whereas S3 is not a protocol specific to AWS, but is also implemented by (eg) Minio, Wasabi, and others. I think it might be better to make these steps more general and less AWS specific.

Also, a sentence explaining the 'push existing data' section would be useful: this only applies if you're moving to s3 from local storage, not if you're setting up GtS for the first time, right?

**Comment by anantshri** [comment]
Hey @tsmethurst makes sense. However I dont have experiance with other S3 environments. this is based on reading other guides doing misconfig then realizing this is far more straightforward then i expected.

I have added markers to make it clear this is AWS S3 specific. May be others can later add details around other implementations.

Reference guides that went all different directions. 
1. https://thomas-leister.de/en/mastodon-s3-media-storage/
2. https://github.com/cybrespace/cybrespace-meta/blob/master/s3.md

File: docs/configuration/storage.md [hunk 1]
```diff
@@ -62,6 +62,60 @@ storage-s3-secret-key: ""
...
+1. Login to AWS -> select S3 as service.
+2. click Create Bucket
+3. Provide a unique name and avoid adding "." in the name
+4. Do not change the public access settings (Let them be on "block public access" mode)
+
+#### AWS ACCESS KEY Configuration
+
+1. In AWS Console -> IAM (under Security, Identity, & Compliance)
+2. Add a user with programatic api's access
+3. You can enable s3full access in case you dont want to care about security but recommendation would be to setup a tight permissions
```
Comment (igalic) [inline docs/configuration/storage.md:80]: or maybe we don't mention how to shoot yourself / your users in the foot?

or actively discourage it?

either way, the don't needs a '

File: docs/configuration/storage.md [hunk 2]
```diff
@@ -62,6 +62,60 @@ storage-s3-secret-key: ""
...
+            "Resource": [
+                "arn:aws:s3:::<bucket_name>",
+                "arn:aws:s3:::<bucket_name>/*"
+            ]
+        }
+    ]
+}
+
+```
+P.S. We gave ListAll for s3cmd to work nicely.
```
Comment (igalic) [inline docs/configuration/storage.md:102]: what does that mean?

File: docs/configuration/storage.md [hunk 3]
```diff
@@ -62,6 +62,60 @@ storage-s3-secret-key: ""
...
+4. Provide the values in config above
+  
+  * storage-s3-endpoint -> should be your bucket location say `s3.ap-southeast-1.amazonaws.com`
+  * storage-s3-access-key -> Access key you obtained for the user created above
+  * storage-s3-secret-key -> Secret key you obtained for the user created above
+  * storage-s3-bucket -> Keep this as the <bucketname> that you created just now.
+
+
+
+#### For Data migration from local storage to AWS s3 bucket in case of running instance. (not required if you are starting fresh)
```
Comment (igalic) [inline docs/configuration/storage.md:112]: this is too long to be a useful header

put the stuff () in a new paragraph, without the (), and a full sentence

**Review by igalic — CHANGES_REQUESTED** [review]
here's some suggestions

**Comment by tsmethurst** [comment]
Thanks @igalic ! 

@anantshri once you've resolved @igalic's suggestions I'm happy to merge this :)

File: docs/configuration/storage.md [hunk 4]
```diff
@@ -62,6 +62,60 @@ storage-s3-secret-key: ""
...
+2. click Create Bucket
+3. Provide a unique name and avoid adding "." in the name
+4. Do not change the public access settings (Let them be on "block public access" mode)
+
+#### AWS ACCESS KEY Configuration
+
+1. In AWS Console -> IAM (under Security, Identity, & Compliance)
+2. Add a user with programatic api's access
+3. You can enable s3full access in case you dont want to care about security but recommendation would be to setup a tight permissions
+replace <bucketname> in this sample policy
```
[the remark this answers is not in the document]
Comment (anantshri) [reply docs/configuration/storage.md:81]: good spot merging this with above line remvoing s3full access part of sentence.

File: docs/configuration/storage.md [hunk 5]
```diff
@@ -62,6 +62,60 @@ storage-s3-secret-key: ""
...
+```
```
[the remark this answers is not in the document]
Comment (anantshri) [reply docs/configuration/storage.md:82]: ah i keep forgetting that. adding.

File: docs/configuration/storage.md (hunk 1 above)
Comment (anantshri) [reply docs/configuration/storage.md:80]: valuid point.

File: docs/configuration/storage.md (hunk 3 above)
Comment (anantshri) [reply docs/configuration/storage.md:112]: Valid point rephrasing.

**Comment by anantshri** [comment]
Updated @igalic please review when you can.

**Review by igalic — APPROVED** [review]
👍

**Comment by tsmethurst** [comment]
Thanks! :)

**Pull request merged**